toffy/mph_comp.py

```diff
@@ -0,0 +1,180 @@
+import os
+import pandas as pd
+import numpy as np
+import json
+import matplotlib.pyplot as plt
+
+from mibi_bin_tools import bin_files
+from ark.utils import io_utils
+
+
+def get_estimated_time(bin_file_dir, fov):
+    """Retrieve run time data for each fov json file
+    Args:
+        bin_file_dir (str): path to the FOV bin and json files
+        fov (str): name of fov to get estimated time for
+    Returns:
+        fov_time (int): estimated run time for the given fov
+    """
+
+    # path validation
+    io_utils.validate_paths(bin_file_dir)
+
+    # get fov json file in bin_file_path
+    json_file = io_utils.list_files(bin_file_dir, fov+".json")
+    if len(json_file) == 0:
+        raise FileNotFoundError(f"The FOV name supplied doesn't have a JSON file: {fov}")
+
+    # retrieve estimated time (frame dimensions x pixel dwell time)
+    with open(os.path.join(bin_file_dir, json_file[0])) as file:
+        run_metadata = json.load(file)
+        try:
+            size = run_metadata.get('frameSize')
+            time = run_metadata.get('dwellTimeMillis')
+            estimated_time = int(size**2 * time)
+        except TypeError:
+            raise KeyError("The FOV json file is missing one of the necessary keys "
+                           "(frameSize or dwellTimeMillis)")
+
+    return estimated_time
+
+
+def compute_mph_metrics(bin_file_dir, csv_dir, fov, mass=98, mass_start=97.5, mass_stop=98.5):
+    """Retrieves total counts, pulse heights, & estimated time for a given FOV
+        Args:
+            bin_file_dir (str): path to the FOV bin and json files
+            csv_dir (str): path to output csv to
+            fov (string): name of fov bin file without the extension
+            mass (float): mass for the panel
+            mass_start (float): beginning of mass integration range
+            mass_stop (float): end of mass integration range
+            """
+
+    target = None
+    panel = pd.DataFrame([{
+        'Mass': mass,
+        'Target': target,
+        'Start': mass_start,
+        'Stop': mass_stop,
+    }])
+
+    # retrieve the data from bin file and output to individual csv
+    pulse_height_file = fov + '-mph_pulse.csv'
+
+    try:
+        median = bin_files.get_median_pulse_height(bin_file_dir, fov,
+                                                   target, panel)
+        count_dict = bin_files.get_total_counts(bin_file_dir, [fov])
+    except FileNotFoundError:
+        raise FileNotFoundError(f"The FOV name supplied doesn't have a JSON file: {fov}")
+
+    count = count_dict[fov]
+    time = get_estimated_time(bin_file_dir, fov)
+
+    out_df = pd.DataFrame({
+        'fov': [fov],
+        'MPH': [median],
+        'total_count': [count],
+        'time': [time]})
+
+    # saves individual .csv files to csv_dir
+    out_df.to_csv(os.path.join(csv_dir, pulse_height_file), index=False)
+
+
+def combine_mph_metrics(csv_dir, return_data=False):
```

Comment:
The MPH specific logic in this function, like getting the cumulative count, can be pulled out into a separate function. Then, you should be able to use this function to do the combination: https://github.com/angelolab/toffy/blob/e15f1d1bb5eea0e139a00d7ecb94092aa3ba2278/toffy/normalize.py#L179

That way all of the watcher-related stuff will have the same code